### PR TITLE
Remove DUM switch and make F90 switch the default

### LINKED ISSUE #354: Remove DUM switch and make F90 the default
Either the DUM or F90 switch is required.  We propose removing support for the “DUM” switch option and making the F90 the default option.   

This work is planned for the end of May 2021.

### PR DESCRIPTION
**Pull request by JessicaMeixner-NOAA**
# Pull Request Summary
Remove the DUM switch and capability and remove the F90 switch by making it the default. 

## Description
The DUM switch was originally implemented due to issues with some of the intrinsic fortran calls, which have not been an issue in a while.  This PR will remove the option for the DUM switch and makes anything behind a F90 switch the default.  This will change every single switch file.  There is no expectation of any answer changes. 

### Issue(s) addressed
*closes #354 

### Testing
* The matrix_ncep was tested with hera.intel.  All tests passed except the normal not b4b tests.  The matrix is not attached since before merging, the F90 switch will need to be re-removed from every switch file and the regression tests will need to be re-run and compared before merging to develop.  At that time, the matrix results will be posted.

### CODE REVIEW

File: regtests/bin/run_test [hunk 1]
```diff
@@ -1925,7 +1925,11 @@ do
       if [ $testST ]
       then #add S T switches 
         \cp -f $path_b/switch $path_b/switch_noST
-        cat $path_b/switch_noST | sed 's/F90/F90 S T /' > $path_b/switch
+        if [ $prog = ww3_grib ]
+          cat $path_b/switch_noST | sed 's/NCEP2/NCEP2 S T /' > $path_b/switch
```
Comment (aliabdolali) [inline regtests/bin/run_test:1930]: I got failure due to the missing `then` here.

**Review by aliabdolali — COMMENTED** [review]
do we need to change model/bin/w3_source:206?  -e 's/$(aPo)\///g' -e 's/SCRIP\///g' -e 's/ftn/F90/g' -e 's/\\//g' \

**Comment by aliabdolali** [comment]
The regtest ran successfully on Hera using intel/impi

known non-b4b cases:
```
mww3_test_03/./work_PR2_UQ_MPI_d2                     (9 files differ)
mww3_test_03/./work_PR2_UNO_MPI_d2                     (8 files differ)
mww3_test_03/./work_PR3_UNO_MPI_e                     (1 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2_c                     (9 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2                     (9 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2_c                     (8 files differ)
mww3_test_03/./work_PR1_MPI_d2                     (10 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2                     (10 files differ)
mww3_test_03/./work_PR3_UNO_MPI_e_c                     (1 files differ)
mww3_test_03/./work_PR2_UNO_MPI_e                     (1 files differ)
mww3_test_07/./work_PR3_UQ                     (3 files differ)
ww3_tp2.10/./work_MPI_OMPH                     (6 files differ)
```
and expexted changes in netcdf headers (changes in the switch):
```
ww3_tic1.4/./work_IC1IS2_1000                     (2 files differ)
ww3_tic1.4/./work_IC2IS2_IC2d                     (2 files differ)
ww3_tic1.4/./work_IC0IS2_1000                     (2 files differ)
ww3_tic1.4/./work_IC2IS2_IC2b                     (2 files differ)
ww3_tic2.2/./work_IC2IS2                     (1 files differ)
ww3_tnc1/./work                     (14 files differ)
ww3_tp2.10/./work_MPI                     (1 files differ)
ww3_tp2.11/./work_MPI                     (1 files differ)
ww3_tp2.14/./work_OASACM5                     (1 files differ)
ww3_tp2.14/./work_OASACM6                     (1 files differ)
ww3_tp2.14/./work_OASACM4                     (1 files differ)
ww3_tp2.14/./work_OASACM3                     (1 files differ)
ww3_tp2.14/./work_OASACM                     (1 files differ)
ww3_tp2.14/./work_OASICM                     (1 files differ)
ww3_tp2.14/./work_OASOCM                     (1 files differ)
ww3_tp2.14/./work_OASACM2                     (1 files differ)
ww3_tp2.15/./work_5km                     (1 files differ)
ww3_tp2.15/./work_PR3_UQ_RHO                     (1 files differ)
ww3_tp2.15/./work_PR3_UQ                     (1 files differ)
ww3_tp2.16/./work_MPI_OMPH                     (2 files differ)
ww3_tp2.16/./work_MPI                     (2 files differ)
ww3_tp2.17/./work_a                     (1 files differ)
ww3_tp2.17/./work_ma1                     (1 files differ)
ww3_tp2.17/./work_mc                     (1 files differ)
ww3_tp2.17/./work_c                     (1 files differ)
ww3_tp2.17/./work_b                     (1 files differ)
ww3_tp2.17/./work_ma                     (1 files differ)
ww3_tp2.17/./work_mb                     (1 files differ)
ww3_tp2.18/./work_TIDE                     (1 files differ)
ww3_tp2.18/./work_TIDE_MPI                     (1 files differ)
ww3_tp2.6/./work_pdlib                     (12 files differ)
ww3_tp2.8/./work_PR3_UQ                     (2 files differ)
ww3_ts1/./work_NL5                     (1 files differ)
ww3_ts1/./work_ST6                     (1 files differ)
ww3_ufs1.1/./work_a_nth                     (1 files differ)
ww3_ufs1.1/./work_a_npl                     (1 files differ)
ww3_ufs1.1/./work_a                     (1 files differ)
ww3_ufs1.1/./work_b                     (1 files differ)
ww3_ufs1.2/./work_a                     (4 files differ)
ww3_ufs1.2/./work_b                     (4 files differ)
```
 


[matrixDiff.txt](https://github.com/NOAA-EMC/WW3/files/6927788/matrixDiff.txt)
[matrixCompSummary.txt](https://github.com/NOAA-EMC/WW3/files/6927789/matrixCompSummary.txt)
[matrixCompFull.txt](https://github.com/NOAA-EMC/WW3/files/6927790/matrixCompFull.txt)

**Comment by JessicaMeixner-NOAA** [comment]
> do we need to change model/bin/w3_source:206? -e 's/$(aPo)///g' -e 's/SCRIP///g' -e 's/ftn/F90/g' -e 's/\//g' \

@aliabdolali we do not use w3_source anymore as we don't make a tarball.  Additionally I do not think this line needs to be change for removing the DUM/F90 switches.

**Comment by aliabdolali** [comment]
> > do we need to change model/bin/w3_source:206? -e 's/$(aPo)///g' -e 's/SCRIP///g' -e 's/ftn/F90/g' -e 's///g' \
> 
> @aliabdolali we do not use w3_source anymore as we don't make a tarball. Additionally I do not think this line needs to be change for removing the DUM/F90 switches.

I see, this is for ftn and F90 files.

**Review by aliabdolali — APPROVED** [review]

**Pull request merged**